requires.py

```diff
@@ -32,16 +32,19 @@ def changed(self):
         conv.toggle_state('{relation_name}.ready',
                           active=self.is_spark_started())
 
-    @hook('{requires:spark}-relation-departed')
-    def departed(self):
-        conv = self.conversation()
-        conv.remove_state('{relation_name}.joined')
-        conv.remove_state('{relation_name}.ready')
```

Comment:
Leaving the states indefinitely will cause other issues.  The best interim work-around I can think of is to test against the remaining units, i.e.:

```python
-    @hook('{requires:spark}-relation-departed')
    def departed(self):
        conv = self.conversation()
        if len(conv.units) == 1: # last remaining unit departing
            conv.remove_state('{relation_name}.joined')
            conv.remove_state('{relation_name}.ready')
```

It's not ideal to have to do this in every interface layer, but I don't see a way around it right now.